### PR TITLE
[Module #3] tasks solved

### PR DESCRIPTION
**Pull request by pbochniak**
* Task 1 in 1st commit
* Task 2 in 2nd (simple own sorting implementation) and 3rd (used sorting from go std library) commits
* Task 3 in 4th commit

### CODE REVIEW

File: module-3/task2/slices/slice.go [hunk 1]
```diff
@@ -1,21 +1,60 @@
...
+	}
+	return false
+}
+
+func (users UserSlice) Swap(i, j int) {
+	users[i], users[j] = users[j], users[i]
+}
+
+func SortUsersPure(users []User) []User {
+	temp_users := make([]User, len(users))
```
Comment (yushni) [inline module-3/task2/slices/slice.go]: Please use camel case. In GO code style we prefer to use camel case under underscore.

File: module-3/task2/slices/slice.go [hunk 2]
```diff
@@ -1,21 +1,60 @@
...
 	FirstName string
 	LastName  string
 }
 
-func SortUsersPure(s []User) []User {
-	return r
+// to implement sort.Interface
+type UserSlice []User
+
+func (users UserSlice) Len() int {
```
Comment (yushni) [inline module-3/task2/slices/slice.go]: I think this is not necessary to create own type that implements `sort.Interface` I'd prefer in this case use sort.Slice

File: module-3/task2/slices/slice.go [hunk 3]
```diff
@@ -1,21 +1,60 @@
...
+	return len(users)
+}
+
+func (users UserSlice) Less(i, j int) bool {
```
Comment (yushni) [inline module-3/task2/slices/slice.go]: My suggestion: 
```

func SortUsersPure(users []User) []User {
	temp_users := make([]User, len(users))
	copy(temp_users, users)

	sort.Slice(temp_users, compareUsers(temp_users))
	return temp_users
}

func SortUsersPureDesc(users []User) []User {
	temp_users := make([]User, len(users))
	copy(temp_users, users)

	sort.Slice(temp_users, func(i, j int) bool {
		return !compareUsers(temp_users)(i, j)
	})
	return temp_users
}

func compareUsers(users []User) func(i, j int) bool {
	return func(i, j int) bool {
		if users[i].FirstName > users[j].FirstName {
			return false
		}
		if users[i].FirstName < users[j].FirstName {
			return true
		}
		if users[i].LastName > users[j].LastName {
			return false
		}
		if users[i].LastName < users[j].LastName {
			return true
		}
		return false
	}
}
```

File: module-3/task3/maps/maps.go [hunk 4]
```diff
@@ -1,4 +1,16 @@
 package maps
 
+import "sort"
+
 func SortedValues(m map[int]string) string {
+	keys := make([]int, 0)
```
Comment (yushni) [inline module-3/task3/maps/maps.go]: If it possible - provide a cap or len `keys := make([]int, 0, len(m))`

**Review by yushni — APPROVED** [review]

File: module-3/task2/slices/slice.go (hunk 1 above)
Comment (pbochniak) [reply module-3/task2/slices/slice.go]: Sure, it just my habit from python (my main language).

**Review by yushni — APPROVED** [review]

**Pull request merged**